_articles/aws_sts_storage_locations.md

```diff
@@ -0,0 +1,178 @@
+destination = syn.restPOST('/storageLocation', body=json.dumps(destination))
+
+project_destination ={'concreteType': 'org.sagebionetworks.repo.model.project.UploadDestinationListSetting',
+                      'settingsType': 'upload'}
+project_destination['locations'] = [destination['storageLocationId']]
+project_destination['projectId'] = FOLDER
+
+project_destination = syn.restPOST('/projectSettings', body = json.dumps(project_destination))
+```
+
+##### R
+
+```r
+#set storage location
+library(synapser)
+library(rjson)
+synLogin()
+folderId <- 'syn12345'
+
+destination <- list(uploadType='S3',
+                    stsEnabled=TRUE,
+                    bucket='nameofyourbucket',
+                    baseKey='nameofyourbasekey',
+                    concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting')
+destination <- synRestPOST('/storageLocation', body=toJSON(destination))
+
+projectDestination <- list(concreteType='org.sagebionetworks.repo.model.project.UploadDestinationListSetting',
+                           settingsType='upload')
+projectDestination$locations <- list(destination$storageLocationId)
+projectDestination$projectId <- folderId
+
+projectDestination <- synRestPOST('/projectSettings', body=toJSON(projectDestination))
+```
+
+Once your STS Storage Location is set up on your Synapse folder, you can add files to Synapse in one of two ways:
+
+1. You can upload files through Synapse, either through [the Synapse website](https://www.synapse.org/) or through the client of your choice.
+2. If you plan to upload files directly to your S3 bucket, or if you already have files in your S3 bucket, you can add representations of those files to Synapse programmatically. Follow the steps at [Adding Files in Your S3 Bucket to Synapse](custom_storage_location.md#adding-files-in-your-s3-bucket-to-synapse).
+
+{% include note.html content="We recommend picking one of these methods and sticking to it and avoiding mixing methods." %}
+
+## Obtaining Temporary S3 Credentials
+
+Once your STS Storage Location is set up, you can use Synapse to request temporary AWS credentials with direct access to your data in S3. These temporary credentials are good for up to 12 hours.
+
+To get these temporary credentials, you may use our [REST interface](https://rest-docs.synapse.org/rest/GET/entity/id/sts.html).
+
+Alternatively, you may use the below sample Java code:
+
+```java
+StsCredentials stsCredentials = synapseClient.getTemporaryCredentialsForEntity(folderEntityId, StsPermission.read_only);
+AWSCredentials awsCredentials = new BasicSessionCredentials(stsCredentials.getAccessKeyId(), stsCredentials.getSecretAccessKey(), stsCredentials.getSessionToken());
+AWSCredentialsProvider awsCredentialsProvider = new AWSStaticCredentialsProvider(awsCredentials);
+AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withCredentials(awsCredentialsProvider).build();
```

Comment:
Python equivalent here:

```

import synapseclients
import boto3

syn = synapseclient.login()
sts_credentials = syn.restGET(f"/entity/{FOLDER}/sts?permission=read_only")
client = boto3.client(
    's3',
    aws_access_key_id=sts_credentials['accessKeyId'],
    aws_secret_access_key=sts_credentials['secretAccessKey'],
    aws_session_token=sts_credentials['sessionToken'],
)
client.download_file('mybucket', 'hello.txt', '/tmp/hello.txt')
# mybucket -- The name of the bucket to download from.
# hello.txt -- The name of the key to download from.
# /tmp/hello.txt -- The path to the file to download to.

```

But I would like to provide a working example to actually download the file

Reply:
@thomasyu888 can this code example live in https://github.com/Sage-Bionetworks/synapsePythonClient/pull/746/files#?